### PR TITLE
Issue #34: Merge Calibrator MCastComponent functionality into Detector

### PR DESCRIPTION
**Pull request by Rampex1**

### CODE REVIEW

**Comment by Rampex1** [comment]
I moved the supported request types from Calibrator to Detector as discussed. However, I still can't remove supported request types in Calibrator without getting the error: "parsable_type did not match any expected value. Input is improperly formatted". I think that some code somewhere that is calling calibrator.supported_request_types should be changed to detector.supported_request_types, but I can't find where it is.

**Comment by vaughantnrc** [comment]
I can't look too deeply right now, but in PyCharm you can try to right-click on the method text, choose "Find References"

**Comment by vaughantnrc** [comment]
I tried to run it just now. When I try to run main_detector it tells me that the calibrator configuration is missing.

Just to check - are you connecting to a detector that is running the code that you edited (I assume that code would not yet be on the Raspberry Pi 5)?

**Comment by Rampex1** [comment]
Calibrator configuration should be working now. 
I am currently connecting to the green detector plugged into my computer.

**Comment by Rampex1** [comment]
I still can't fix the issue. The closest I've come to making it work is changing the connection label in self._connector.request_series_push() from selected_calibrator_label to selected_detector_label. This removes the "wrong parsable type" error but then creates more bugs in the calibrator panel.

**Comment by vaughantnrc** [comment]
Hi David, to clarify, the goal is to migrate all the calibrator functionality into the detector, and thereby remove the need for a separate calibrator process. I worked a bit more toward that in the branch here: https://github.com/vaughantnrc/MCSTrack/tree/merge_calibrator_into_detector

I haven't had time to test, but please take a look at the code and pull these changes into your branch.

File: src/detector/detector.py [hunk 1]
```diff
@@ -15,6 +15,9 @@
     StartCaptureRequest, \
     StopCaptureRequest
 from src.detector.fileio import DetectorConfiguration
+from src.calibrator import Calibrator
+from src.calibrator.fileio import CalibratorConfiguration
+from src.calibrator.api import *
```
Comment (vaughantnrc) [inline src/detector/detector.py:20]: Generally it's best to spell out exactly what you're importing; It'll be easier to find where those classes are used.

File: src/gui/panels/detector_panel.py [hunk 2]
```diff
@@ -822,17 +799,14 @@ def begin_set_detection_parameters(self):
         self.disable_parameter_controls()
 
     def disable_calibrator_controls(self):
-        self.set_calibrator_controls_enabled(enable=False)
+        pass
```
Comment (vaughantnrc) [inline src/gui/panels/detector_panel.py:802]: You might be able to remove this function entirely (and all calls to it)?

**Review by vaughantnrc — APPROVED** [review]
Looks good to me. :+1: Please go ahead and merge when ready. If you want to address my comments then please do so. No need to squash (it'll just be 2 commits so I consider it to be fine).

**Pull request merged**